### PR TITLE
Make TypedConstantRewiter work with nested structs

### PR DESCRIPTION
**Pull request by ptomin**
- `TypedConstantRewiterTests`: simplify `Given_Global`
Do not subtract `0x100000` from address
-  Create `TypedConstantRewriter` test with pointer to nested struct field
Expected:
```
    &globals->t100100.r0004
```
But was:
```
    &globals->dw100104
```
- Make `TypedConstantRewiter` work with complex global variables(e.g. structs)
- `ComplexExpressionBuilder.FallbackExpression`: avoid exception if data type size is unknown

### CODE REVIEW

File: subjects/PE/x86/VCExeSample/VCExeSample.reko/VCExeSample_text.c [hunk 1]
```diff
@@ -125,10 +125,10 @@ void gbl_nested_structs_test14()
...
 	globals->gbl_nested_structs = (Eq_n) 0x05;
 	// gbl_nested_structs.str.b = 6
-	globals->t403020 = (Eq_n) 0x06;
+	globals->gbl_nested_structs.str.b = (Eq_n) 0x06;
 	// gbl_nested_structs.str.c = 7
-	globals->dw403024 = 0x07;
+	globals->gbl_nested_structs.str.c = 0x07;
 	// gbl_nested_structs.d = 8
-	globals->dw403028 = 0x08;
+	globals->gbl_nested_structs.d = 0x08;
```
Comment (ptomin) [inline subjects/PE/x86/VCExeSample/VCExeSample.reko/VCExeSample_text.c:132]: Look better, is not it?

File: src/Decompiler/Typing/TypedConstantRewriter.cs [hunk 2]
```diff
@@ -365,6 +363,39 @@ DataType PromoteToCString(Constant c, DataType charType)
             return dt;
         }
 
+        private bool IsInsideField(int offset, StructureField field)
+        {
+            if (offset == field.Offset)
+                return true;
+            if (offset < field.Offset + field.DataType.Size)
+                return true;
+            var str = field.DataType.ResolveAs<StructureType>();
```
Comment (uxmal) [inline src/Decompiler/Typing/TypedConstantRewriter.cs:372]: Should this not be:
```
if (field.Offset <= offset && offset < field.Offset + field.DatType.Size)
```

Comment (ptomin) [reply src/Decompiler/Typing/TypedConstantRewriter.cs:374]: `StructureFieldCollection.LowerBound`  guarantees that `field.Offset <= offset` is always `true`. But I'll add additional check for safety

File: src/Decompiler/Typing/TypedConstantRewriter.cs [hunk 3]
```diff
@@ -365,6 +363,41 @@ DataType PromoteToCString(Constant c, DataType charType)
             return dt;
         }
 
+        private bool IsInsideField(int offset, StructureField field)
+        {
+            if (offset < field.Offset)
+                return false;
```
Comment (ptomin) [inline src/Decompiler/Typing/TypedConstantRewriter.cs:369]: John, is this check enough?

Comment (uxmal) [reply src/Decompiler/Typing/TypedConstantRewriter.cs:369]: Let's go with this.

**Pull request merged**